src/cplus_plugin/lib/reports/generator.py

```diff
@@ -1798,6 +1798,180 @@ def _update_main_map_legend(self):
         legend_item.invalidateCache()
         legend_item.update()
 
+    def _filter_impact_matrix_table(
+        self,
+        pathway_ids: typing.List[str] = None,
+        priority_layer_ids: typing.List[str] = None,
+    ) -> dict:
+        """Filter the impact matrix table by specific NCS pathway and/or
+        priority layer IDs.
+
+        :param pathway_ids: List of pathway UUIDs to filter or None to keep all.
+        :type pathway_ids: typing.List[str]
+        :param priority_layer_ids: List of priority layer UUIDs to filter or
+        None to keep all.
+        :type priority_layer_ids: typing.List[str]
+
+        :returns: Filtered dictionary with the same format as that used for the
+        impact matrix stored in settings
+        i.e. {pathway_uuids: [], priority_layer_uuids: [], values: [[], []...]}
+        :rtype: dict
+        """
+        impact_matrix_str = settings_manager.get_value(
+            Settings.SCENARIO_IMPACT_MATRIX, "{}"
+        )
+        try:
+            impact_matrix = json.loads(impact_matrix_str) or {}
+        except json.JSONDecodeError:
+            return {}
+
+        if not impact_matrix:
+            return {}
+
+        # Get indices for pathways to keep
+        if pathway_ids is None:
+            pathway_indices = list(range(len(impact_matrix["pathway_uuids"])))
```

Comment:
Please add checking to prevent `KeyError`, unless `SCENARIO_IMPACT_MATRIX` has been validated before save